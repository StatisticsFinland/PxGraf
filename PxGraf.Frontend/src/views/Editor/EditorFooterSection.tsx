import React from 'react';
import { Box, Button } from '@mui/material';
import SaveIcon from '@mui/icons-material/Save';
import { EditorContext } from 'contexts/editorContext';
import { useTranslation } from 'react-i18next';
import styled from 'styled-components';
import InfoBubble from 'components/InfoBubble/InfoBubble';

const FooterBtnWrapper = styled(Box)`
    grid-area: 'footer';
    display: flex;
    justify-content: flex-end;
    align-items: center;
    padding: 16px;
`;


export const EditorFooterSection: React.FC = () => {
    
    const { setSaveDialogOpen } = React.useContext(EditorContext);
    
    const { t } = useTranslation();
    
    return(
        <FooterBtnWrapper>
            <InfoBubble info={t('infoText.save')} ariaLabel={t("editor.save")} />
            <Button variant="contained" startIcon={<SaveIcon />} onClick={() => setSaveDialogOpen(true)}>
                {t("editor.save")}
            </Button>
        </FooterBtnWrapper>
    );
}

export default EditorFooterSection;