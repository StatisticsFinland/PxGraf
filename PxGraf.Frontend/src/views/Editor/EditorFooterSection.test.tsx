import { render } from '@testing-library/react';
import React from 'react';
import EditorFooterSection from './EditorFooterSection';

jest.mock('react-i18next', () => ({
    ...jest.requireActual('react-i18next'),
    useTranslation: () => {
        return {
            t: (str: string) => str,
            i18n: {
                changeLanguage: () => new Promise(() => {}),
            },
        };
    },
}));

describe('Rendering test', () => {
    it('renders correctly', () => {
        const { asFragment } = render(<EditorFooterSection />);
        expect(asFragment()).toMatchSnapshot();
    });
});
