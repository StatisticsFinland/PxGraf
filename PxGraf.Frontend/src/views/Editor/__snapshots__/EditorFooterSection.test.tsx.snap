// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering test renders correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-fQpRED bwRnBJ MuiBox-root css-0"
  >
    <button
      aria-label="tooltip.open: editor.save"
      class="sc-braxZu jCpbQg"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-colorInfo MuiSvgIcon-fontSizeMedium css-1yq1soz-MuiSvgIcon-root"
        data-testid="InfoIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m1 15h-2v-6h2zm0-8h-2V7h2z"
        />
      </svg>
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-sghohy-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="SaveIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M17 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V7zm-5 16c-1.66 0-3-1.34-3-3s1.34-3 3-3 3 1.34 3 3-1.34 3-3 3m3-10H5V5h10z"
          />
        </svg>
      </span>
      editor.save
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
